gdbgui/src/js/BinaryLoader.jsx

```diff
@@ -28,16 +29,44 @@ class BinaryLoader extends React.Component {
             this.state.past_binaries = []
         }
     }
+    set_button_as_load_binary(){
+        this.setState({load_binary: true})
+        document.getElementById("loadbutton").innerHTML = "Load Button";
+        document.getElementById("binary").placeholder = "/path/to/target/executable -and -flags";
+    }
+    set_button_as_attach_process(){
+        this.setState({load_binary: false})
+        document.getElementById("loadbutton").innerHTML = "Attach Process";
+        document.getElementById("binary").placeholder = "Process id";
+    }
+    click_action_of_button() {
+        if(this.state.load_binary){
+            this.click_set_target_app()
+        } else {
+            this.click_set_target_attach()
+        }
+    }
     render(){
         return(
                 <form style={{marginBottom: 1, flex: '2 0 0'}}>
                   <div className="input-group input-group-sm">
                     <span className="input-group-btn">
-                      <button
+                      <button id="loadbutton"
```

Comment:
a really cool benefit of using React is that you don't have to know anything about the DOM, for the most part.

Instead of using `id`s, you can use the state of the react component. So when rendering this component, you should check the state to see if it's `load_binary` or not (side note: I would probably recommend changing this to `desired_load_type` with options `binary` or `process`). If it's load binary, you can render the `Load Binary` button, otherwise render the `Attach Process` button. You can create variables with the various buttons, like `let load_binary_button = ...` where `...` is valid JSX. Then you can use that with `{load_binary_button}`.

One should never have to call `document.getelementById` when using a React component.

Reply:
Thanks for that tip, I am not very familiar with React or even with Javascript for that matter. I will revisit and clean this code.

Reply:
Thanks for being so open minded and trying to fix it. If you want, I can take what you've done and finish it up myself.

Reply:
I have done the following fixes:
1) changed class to className
2) Got  the dropdown button on the left side
3) Dropdown appears like options ( but still it needs improvement in its appearance)

I still need to :
1) Recheck about inserting the breakpoint at main
2) eliminate document.getelementById

I will give this a try today and submit, and then maybe if its ok with you you can take it forward.